bpf/common/trace_common.h

```diff
@@ -108,14 +108,29 @@ find_nginx_parent_trace(const pid_connection_info_t *p_conn, u16 orig_dport) {
     return NULL;
 }
 
-static __always_inline const tp_info_pid_t *find_nodejs_parent_trace() {
+static __always_inline const tp_info_pid_t *
+find_nodejs_parent_trace(const pid_connection_info_t *p_conn, u16 orig_dport) {
+    connection_info_part_t client_part = {};
+    populate_ephemeral_info(&client_part, &p_conn->conn, orig_dport, p_conn->pid, FD_CLIENT);
+    fd_info_t *fd_info = fd_info_for_conn(&client_part);
+
+    if (!fd_info) {
+        return NULL;
+    }
+
     const u64 pid_tgid = bpf_get_current_pid_tgid();
-    const s32 *node_parent_request_fd = bpf_map_lookup_elem(&active_nodejs_ids, &pid_tgid);
+    const u64 client_key = (pid_tgid << 32) | fd_info->fd;
```

Comment:
What's the intention for the tgid to be used as a key? I think it might need to be the PID included here, rather than just the thread id? I guess if the tgid is unique then it would work, but I thought it isn't.